src/super_gradients/training/transforms/processing.py

```diff
@@ -0,0 +1,186 @@
+from typing import Tuple, List, Union
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import numpy as np
+
+from super_gradients.training.transforms.utils import (
+    _rescale_image,
+    _rescale_bboxes,
+    _pad_image_on_side,
+    _get_center_padding_params,
+    _shift_image,
+    _shift_bboxes,
+)
+
+
+@dataclass
+class ProcessingMetadata(ABC):
+    """Metadata including information to postprocess a prediction."""
+
+
+@dataclass
+class ComposeProcessingMetadata(ProcessingMetadata):
+    metadata_lst: List[Union[None, ProcessingMetadata]]
+
+
+@dataclass
+class DetectionPadToSizeMetadata(ProcessingMetadata):
+    shift_w: float
+    shift_h: float
+
+
+@dataclass
+class RescaleMetadata(ProcessingMetadata):
+    original_size: Tuple[int, int]
+    sy: float
```

Comment:
1) Please add the docstring to this class that explains meaning of sx/sy parameters - are they represent original_size / preprocessed_size or vice-versa?
2) original_size is rows,cols or width,height? This should be also explicitly stated everywhere to avoid confusion of users. For sake of consistency I suggest to use term "shape" for rows,cols and "size" for width, height.

Reply:
1. I renamed to `scale_factor_h` and `scale_factor_w`. I believe this is now self-explanatory, do you agree or still prefer a docstring?
2. I agree, I actually forgot to do this cleanup thanks for noticing. Just to make sure, when you write (rows,cols), you means (n_rows, n_cols) right ? Which is equivalent to (H, W) and is basically how it is stored in np.ndarray.